arbos/arbosState/arbosstate.go

```diff
@@ -189,6 +194,11 @@ func InitializeArbosState(stateDB vm.StateDB, burner burn.Burner, chainConfig *p
 		return nil, errors.New("cannot initialize to ArbOS version 0")
 	}
 
+	serializedChainConfig, err := json.Marshal(chainConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to serialize chain config")
```

Comment:
We shouldn't reserialize the chain config here because it might change. We'll need to pass in the string chain config from the init message, which we can easily get in the replay binary. For the main node, we'll need to create a new method to read the init message without the inbox reader. We should be able to do that by using `delayed.go` to read delayed message 0 from the block where the rollup was created (which is in the config).